case-lib/hijack.sh

```diff
@@ -72,7 +72,8 @@ function func_exit_handler()
 
             local nlines; nlines=$(wc -l < "$logfile") # line count only
             # The first line is the sof-logger header
-            if [ "$nlines" -le 1 ]; then
+            # Don't override exit_status if already SKIPped test case
+            if ([ "$nlines" -le 1 ] && [ $exit_status -ne 2 ]); then
```

Comment:
I really don't get the flow in this file. If an error was already identified (exist_status=1), then what is the point of trying to check additional things?

In this case, if the exit_status is already not zero, why add something related to an empty logger trace?

Reply:
agree, flow is not clean here. exit_status is overwritten under many conditions in this hijack function. Original value of exit_status is easily forgotten.